tests/dataplicity/m2m/test_bencode.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8
+from dataplicity.m2m.bencode import encode, decode, EncodingError, DecodeError
+import pytest
+
+
+def test_bencode_encoder():
+    """ test bencoding capabilities
+    """
+    assert encode({}) == b'de'  # empty dict
+    assert encode({b'foo': 'bar'}) == b'd3:foo3:bare'
+    assert encode({
+        b'foo': 'bar',
+        b'fooo': 'bbar'
+    }) == b'd3:foo3:bar4:fooo4:bbare'
+    assert encode([]) == b'le'  # empty list
+    assert encode(()) == b'le'  # empty list
+    assert encode([1, 2, 3]) == b'li1ei2ei3ee'
+    assert encode([1, 'foo']) == b'li1e3:fooe'
+    assert encode(1) == b'i1e'
+    # make sure that the numbers are encoded as bytes. This line would always
+    # return true in python2, but will have a meaning in python3
+    assert isinstance(encode(1), bytes)
+    # utf-8 string
+    assert encode(b'a\xc5\xbc') == b'3:a\xc5\xbc'
+    assert encode("aż") == b"3:a\xc5\xbc"
+    with pytest.raises(EncodingError):
+        encode(1.38)
+    # since bytes == str in python2, this exception won't be raised
+    with pytest.raises(EncodingError):
+        encode({u'foo': 'bar'})
+
+    assert encode(-41) == b"i-41e"
+
+
+def test_bencode_decoder():
+    """ test decoding capabilities of bencode module
+    """
+    # portable (python2+3) way of testing whether object is instance
+    # of bytes ;)
+    with pytest.raises(Exception) as exc:
```

Comment:
Its dubious if you need to catch anything at all. There's no recovering from decoding the wrong object, the assert was to catch screw-ups during development.

Putting that in a test just enforces the AssertionError it as part of the api, which it isn't really. Best remove that test, since the test will never really detect any issues...